kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java

```diff
@@ -427,27 +430,29 @@ private void wrapValue(final AssetRecord assetRecord, final String userValue, fi
         requireNonNull(userValue, message.valueNonNull());
         requireNonNull(userType, message.typeNonNull());
 
+        final DataType dataType = DataType.getDataType(userType);
```

Comment:
Add this inside the `try` block as it might throw `IllegalArgumentException` and the exception must be propagated upwards. Also catch `IllegalArgumentException` in this method as well as in `doPutAssets()` in the cloudlet.